supplier/api/build.gradle.kts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.nosphere.apache.rat.RatTask
+
+plugins {
+    id("java-library")
+    id("maven-publish")
+    id("signing")
+    id("org.nosphere.apache.rat") version "0.4.0"
+}
+
+// -----------------------------------------------------------------------------
+// Dependencies
+// -----------------------------------------------------------------------------
+dependencies {
+    api("io.knotx:knotx-fragment-api:${project.version}")
+    api("io.knotx:knotx-server-http-api:${project.version}")
+
+    annotationProcessor(platform("io.knotx:knotx-dependencies:${project.version}"))
+    annotationProcessor(group = "io.vertx", name = "vertx-codegen")
+    annotationProcessor(group = "io.vertx", name = "vertx-service-proxy", classifier = "processor")
+    annotationProcessor(group = "io.vertx", name = "vertx-rx-java2-gen")
+
+    implementation(group = "org.apache.commons", name = "commons-lang3")
+}
+
+// -----------------------------------------------------------------------------
+// Source sets
+// -----------------------------------------------------------------------------
+tasks.named<JavaCompile>("compileJava") {
+    options.annotationProcessorGeneratedSourcesDirectory = file("src/main/generated")
+}
+tasks.named<Delete>("clean") {
+    delete.add("src/main/generated")
+}
+sourceSets.named("main") {
+    java.srcDir("src/main/generated")
+}
+
+// -----------------------------------------------------------------------------
+// Tasks
+// -----------------------------------------------------------------------------
+tasks {
+    named<RatTask>("rat") {
+        excludes.addAll("**/*.md", "**/*.adoc", "**/build/*", "**/out/*", "**/generated/*")
+    }
+    getByName("build").dependsOn("rat")
+}
+
+// -----------------------------------------------------------------------------
+// Publication
+// -----------------------------------------------------------------------------
+tasks.register<Jar>("sourcesJar") {
+    from(sourceSets.named("main").get().allJava)
+    classifier = "sources"
+}
+tasks.register<Jar>("javadocJar") {
+    from(tasks.named<Javadoc>("javadoc"))
+    classifier = "javadoc"
+}
+tasks.named<Javadoc>("javadoc") {
+    if (JavaVersion.current().isJava9Compatible) {
+        (options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
+    }
+}
+
+publishing {
```

Comment:
Can you use Knot.x Gradle Plugin? Please see https://github.com/Knotx/knotx-server-http/pull/29.